SmartDeviceLink/SDLIAPTransport.m

```diff
@@ -88,19 +89,25 @@ - (void)sdl_stopEventListening {
 #pragma mark - EAAccessory Notifications
 
 - (void)sdl_accessoryConnected:(NSNotification *)notification {
+    EAAccessory *accessory = [notification.userInfo objectForKey:EAAccessoryKey];
     NSMutableString *logMessage = [NSMutableString stringWithFormat:@"Accessory Connected, Opening in %0.03fs", self.retryDelay];
     [SDLDebugTool logInfo:logMessage withType:SDLDebugType_Transport_iAP toOutput:SDLDebugOutput_All toGroup:self.debugConsoleGroupName];
 
     self.retryCounter = 0;
-    [self performSelector:@selector(connect) withObject:nil afterDelay:self.retryDelay];
+
+    [self performSelector:@selector(sdl_connect:) withObject:accessory afterDelay:self.retryDelay];
 }
 
 - (void)sdl_accessoryDisconnected:(NSNotification *)notification {
     [SDLDebugTool logInfo:@"Accessory Disconnected Event" withType:SDLDebugType_Transport_iAP toOutput:SDLDebugOutput_All toGroup:self.debugConsoleGroupName];
+    [self sdl_stopEventListening];
 
     // Only check for the data session, the control session is handled separately
     EAAccessory *accessory = [notification.userInfo objectForKey:EAAccessoryKey];
-    if (accessory.connectionID == self.session.accessory.connectionID) {
+    if (accessory.connectionID != self.session.accessory.connectionID) {
+         [SDLDebugTool logInfo:@"Accessory connection ID mismatch!!!" withType:SDLDebugType_Transport_iAP toOutput:SDLDebugOutput_All toGroup:self.debugConsoleGroupName];
+    }
+    if ([accessory.name isEqualToString:self.session.accessory.name]){
```

Comment:
The idea is to guard against another accessory disconnect, say for an MFi BT headset, causing a false positive disconnect in the SDL app. I'm happy to base this on serial number, but I do strongly recommend that if we think our accessory has gone away that we clean up regardless of the connection ID.